base/reduce.jl

```diff
@@ -954,30 +956,30 @@ julia> count([true, false, true, true])
 3
 ```
 """
-count(itr) = count(identity, itr)
+count(itr; init=0) = count(identity, itr; init)
 
-count(f, itr) = _simple_count(f, itr)
+count(f, itr; init=0) = _simple_count(f, itr, init)
 
-function _simple_count(pred, itr)
-    n = 0
+function _simple_count(pred, itr, init)
+    n = init
     for x in itr
         n += pred(x)::Bool
     end
     return n
 end
 
-function count(::typeof(identity), x::Array{Bool})
-    n = 0
+function _simple_count(::typeof(identity), x::Array{Bool}, init::T=0) where {T}
+    n::T = init
     chunks = length(x) ÷ sizeof(UInt)
     mask = 0x0101010101010101 % UInt
     GC.@preserve x begin
         ptr = Ptr{UInt}(pointer(x))
         for i in 1:chunks
-            n += count_ones(unsafe_load(ptr, i) & mask)
+            n += count_ones(unsafe_load(ptr, i) & mask) % T
```

Comment:
I think I would rather write 
```suggestion
            n = (n + count_ones(unsafe_load(ptr, i) & mask)) % T
```
for the case where `T` is more heavy to construct, e.g. `BigInt`. Conversely, if `T` is smaller than `Int`, it probably doesn't matter to promote the addition to `Int` before the `%`-reduction. Similarly for `BitArray` and below.
But this is just a suggestion, which is maybe too defensive.